### PR TITLE
Add optional 'Next of Kin' fields to registration page

### PR DESCRIPTION
**Pull request by DoxasticFox**
Resolves https://github.com/Australian-Genomics/CTRL/issues/57.

### CODE REVIEW

File: app/lib/next_of_kin_registration_validator.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module NextOfKinRegistrationValidator
+  def next_of_kin_needed_to_register?
+    # TODO: Put survey config in in yaml file
```
Comment (DoxasticFox) [inline app/lib/next_of_kin_registration_validator.rb:3]: This depends on https://github.com/Australian-Genomics/CTRL/pull/56 being merged.

**Review by mathob — COMMENTED** [review]
It seems a little odd to me that the next of kin phone is a required field but the subject's phone is not.

Also while looking at the personal details edit page I notice that :

- the subject can choose preferred contact method of phone/email/mail but these fields could all be empty - shouldn't selecting e.g. preferred contact method "phone" then mean that the phone field is required
- there's no check that a phone number is valid (can type anything in there at the moment)

So is it worthwhile adding some user input validation stuff here?

**Comment by DoxasticFox** [comment]
>It seems a little odd to me that the next of kin phone is a required field but the subject's phone is not.

You're right. I think I thoughtlessly copied/pasted that from the personal details edit page. Fix it now...

The other points you make are good too. I raised a ticket (https://github.com/Australian-Genomics/CTRL/issues/59).

**Review by mathob — COMMENTED** [review]
With '"Next of Kin" Needed to Register' set to false in the admin, nevertheless when a user attempts to edit personal details the next of kin first name, family name and contact phone fields are all required ("Can't be blank")

**Comment by DoxasticFox** [comment]
@rosiejbrown What should we do about the quirk @mathob identified? Is it even a quirk? The "Next of Kin" were required to edit the user's personal details even before I added them to the registration page. But is that what we want?

**Comment by rosiejbrown** [comment]
@DoxasticFox Just to clarify, does this mean that when the NOK is switched 'on' (True) it appears in the registration page and is a compulsory field. When it is 'off' (false) the NOK part doesn't appear on the registration page, but is still a compulsory component of the 'extra personal information' they are required to enter off of the dashboard?
If so I think this is correct, the NOK should probably still be compulsory on the extra personal details page.

**Comment by DoxasticFox** [comment]
@rosiejbrown Yep, your understanding is correct.

**Review by mathob — APPROVED** [review]
This is working as expected

**Pull request merged**